### PR TITLE
Fnet training

### PR DESCRIPTION
**Pull request by gregjohnso**
Check's final updates before he left + updated readme from @calystay

### CODE REVIEW

File: examples/download_and_train.py [hunk 1]
```diff
@@ -93,7 +96,7 @@
 with open(prefs_save_path, "w") as fp:
     json.dump(prefs, fp)
 
-command_str = "fnet train {} --gpu_ids {}".format(prefs_save_path, gpu_id)
+command_str = "fnet train --json {} --gpu_ids {}".format(prefs_save_path, gpu_id)
```
Comment (JacksonMaxfield) [inline examples/download_and_train.py]: fstrings are cool / I think they are even more legible. 🤷‍♀ 

```suggestion
command_str = f"fnet train --json {prefs_save_path} --gpu_ids {gpu_id}"
```

File: examples/download_and_train.py [hunk 2]
```diff
@@ -10,7 +11,7 @@
 # Download the 3D multi-channel tiffs via Quilt/T4
 ###################################################
 
-gpu_id = 0
+gpu_id = 7
```
Comment (JacksonMaxfield) [inline examples/download_and_train.py]: Unless the actual fnet application code has drastically changed, I believe this should always be zero for most users.

```suggestion
gpu_id = 0
```

File: fnet/cli/init.py [hunk 3]
```diff
@@ -23,75 +23,133 @@ def save_example_scripts(path_save_dir: str) -> None:
     if not os.path.exists(path_save_dir):
         os.makedirs(path_save_dir)
     path_examples_dir = os.path.join(
-        os.path.dirname(sys.modules['fnet'].__file__), 'cli'
+        os.path.dirname(sys.modules["fnet"].__file__), "cli"
```
Comment (JacksonMaxfield) [inline fnet/cli/init.py]: I know you didn't write this code but this is hard to parse.

File: fnet/cli/predict.py [hunk 4]
```diff
@@ -156,7 +156,7 @@ def parse_model(model_str: str) -> Dict:
 
 
 def save_predictions_csv(
-        path_csv: Path,
+        path_csv: str,
```
Comment (JacksonMaxfield) [inline fnet/cli/predict.py]: Curious, we went from a path type to a string? Any reason?

File: fnet/cli/predict.py [hunk 5]
```diff
@@ -200,89 +210,66 @@ def save_args_as_json(args: argparse.Namespace) -> None:
...
+    """Predicts using model."""
     if args is None:
         parser = argparse.ArgumentParser()
         add_parser_arguments(parser)
         args = parser.parse_args()
-    path_pred_csv = Path(args.path_save_dir, 'predictions.csv')
-    if path_pred_csv.exists():
-        LOGGER.info(f'Using existing prediction results: {path_pred_csv}')
-        return aggregate_results(path_pred_csv, args.metric)
+    if (args.json is not None) and (not args.json.exists()):
```
Comment (JacksonMaxfield) [inline fnet/cli/predict.py]: Can be shortened to:

```suggestion
    if args.json and not args.json.exists():
```

File: fnet/cli/train_model.py [hunk 6]
```diff
@@ -92,27 +92,27 @@ def main(args: Optional[argparse.Namespace] = None) -> None:
         parser = argparse.ArgumentParser()
         add_parser_arguments(parser)
         args = parser.parse_args()
-    if not os.path.exists(args.json):
+    if not args.json.exists():
```
Comment (JacksonMaxfield) [inline fnet/cli/train_model.py]: Do you want to add the same `if args.json and not args.json.exists()` here as you did in the `predict.py`?

```suggestion
    if args.json and not args.json.exists():
```

File: fnet/cli/predict.py (hunk 4 above)
Comment (gregjohnso) [reply fnet/cli/predict.py]: It seems like it got changed to Path for the associated utilities, i.e. path.exists()
